From d38fd3771e5dcf77c2c44e048fa2dc3ade84f454 Mon Sep 17 00:00:00 2001
From: Ben Peled <bpeled@marvell.com>
Date: Mon, 3 Dec 2018 12:18:35 +0200
Subject: [PATCH 0225/1239] arm64: mvebu: dts: mcbin: add support Single Shot
 variant

MacchiatoBin single shot board doesn't have
two 3310 PHY's connected to port 0 on each CP

Change-Id: I48327a1b6e4909cf26a325d4c9d9703723995f81
Signed-off-by: Ben Peled <bpeled@marvell.com>
---
 arch/arm/dts/Makefile                         |  1 +
 .../arm/dts/armada-8040-mcbin-single-shot.dts | 28 +++++++++++++++++++
 configs/mvebu_mcbin-88f8040_defconfig         |  1 -
 3 files changed, 29 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm/dts/armada-8040-mcbin-single-shot.dts

diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index a2be46e820..cf65d8c65d 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -208,6 +208,7 @@ dtb-$(CONFIG_ARCH_MVEBU) +=			\
 	armada-8040-db-G.dtb			\
 	armada-8040-db-H.dtb			\
 	armada-8040-mcbin.dtb			\
+	armada-8040-mcbin-single-shot.dtb	\
 	armada-8040-clearfog-gt-8k.dtb		\
 	armada-8040-ocp.dtb			\
 	armada-8040-ucpe.dtb			\
diff --git a/arch/arm/dts/armada-8040-mcbin-single-shot.dts b/arch/arm/dts/armada-8040-mcbin-single-shot.dts
new file mode 100644
index 0000000000..44135daba7
--- /dev/null
+++ b/arch/arm/dts/armada-8040-mcbin-single-shot.dts
@@ -0,0 +1,28 @@
+/*
+ * Copyright (C) 2018 Marvell International Ltd.
+ *
+ * SPDX-License-Identifier:	GPL-2.0
+ * https://spdx.org/licenses
+ */
+
+#include "armada-8040-mcbin.dts"
+
+/{
+	model = "Marvell 8040 MACHIATOBin SingleShot";
+	compatible = "marvell,armada8040-mcbin-single-shot",
+		     "marvell,armada8040-mcbin",
+		     "marvell,armada8040";
+};
+
+/* 10G SFI/copper */
+&cp0_eth0 {
+	/delete-property/phy;
+};
+
+/* 10G SFI/copper */
+&cp1_eth0 {
+	/delete-property/phy;
+};
+
+/delete-node/ &sfi_phy0;
+/delete-node/ &sfi_phy8;
diff --git a/configs/mvebu_mcbin-88f8040_defconfig b/configs/mvebu_mcbin-88f8040_defconfig
index 2ba7d013d3..97cb642d2b 100644
--- a/configs/mvebu_mcbin-88f8040_defconfig
+++ b/configs/mvebu_mcbin-88f8040_defconfig
@@ -38,7 +38,6 @@ CONFIG_CMD_MVEBU_PHY_FW_DOWNLOAD=y
 CONFIG_CMD_REGULATOR=y
 CONFIG_CMD_EXT4_WRITE=y
 CONFIG_MAC_PARTITION=y
-CONFIG_DEFAULT_DEVICE_TREE="armada-8040-mcbin"
 CONFIG_ENV_IS_IN_SPI_FLASH=y
 CONFIG_NET_RANDOM_ETHADDR=y
 CONFIG_AHCI_MVEBU=y
-- 
2.29.0

